src/core/public/chrome/global_search/global_search_service.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ReactNode } from 'react';
+
+export enum SearchObjectTypes {
+  PAGES = 'pages',
+  SAVED_OBJECTS = 'saved_objects',
+}
+
+/**
+ * @experimental
+ */
+export interface GlobalSearchCommand {
+  /**
+   * unique id of this strategy
+   */
+  id: string;
+  /**
+   * search object type
+   * @type {SearchObjectTypes}
+   */
+  type: SearchObjectTypes;
```

Comment:
Seems the type here is useless, all the search / navigation logic is within `run` callback actually.